test/feedretriever.cpp

```diff
@@ -55,14 +57,36 @@ TEST_CASE("Feed retriever adds header with etag info if available", "[FeedRetrie
 		{"content-type", "text/xml"},
 	}, {});
 
-	const auto feed = feedRetriever.retrieve(url);
+	REQUIRE_THROWS_AS(feedRetriever.retrieve(url), rsspp::NotModifiedException);
 	REQUIRE(testServer.num_hits(mockRegistration) == 1);
-	REQUIRE(feed.items.size() == 0);
 }
 
-TEST_CASE("Feed retriever does not retry download on HTTP 304 (Not Modified)",
+TEST_CASE("Feed retriever retries download if no data is received",
+	"[FeedRetriever]")
+{
+	ConfigContainer cfg;
+	Cache rsscache(":memory:", &cfg);
+	CurlHandle easyHandle;
+	FeedRetriever feedRetriever(cfg, rsscache, easyHandle);
+
+	auto& testServer = test_helpers::HttpTestServer::get_instance();
+	const auto address = testServer.get_address();
+	const auto url = strprintf::fmt("http://%s/feed", address);
+
+	auto mockRegistration = testServer.add_endpoint("/feed", {}, 200, {
+		{"content-type", "text/xml"},
+	}, {});
+
+	cfg.set_configvalue("download-retries", "3");
```

Comment:
`download-retries` applies in surprisingly few cases.
It does not apply if:
- an HTTP code in the 4xx or 5xx range is received. That would throw `rsspp::Exception` (indirectly configured via `CURLOPT_FAILONERROR`).
- there is some data in the request. In that case, XML parsing fails with an exception containing `"XML root node is NULL"` or a specific atom/rss parser throws a parsing error.
- an HTTP status indicating success is received. That would lead to normal parsing of the feed which either results in a valid feed or in an exception (which would avoid download retrying)

Reply:
That seems okay. As far as I understand, `download-retries` is used to overcome very transient errors caused by spotty or extremely slow networks. The list above contains somewhat-prolonged failures, which warrant waiting a bit, so it's fine to ignore them during the reload in the hopes that they'll go away before the next reload.